server/chat/backend/agent/workflow.py

```
@@ -1335,6 +1316,32 @@ async def stream(self, input_state: State):
                             self._last_state.update(stream_data)
                 
             logger.info(f"[WORKFLOW STREAM] Completed: {_event_count} events, {_token_count} tokens streamed")
+
+            # Check if guardrails blocked the message during agentic_tool_flow
+            if _token_count == 0 and _guardrail_task is not None:
+                # The guardrail task ran concurrently; if it blocked, agentic_tool_flow
+                # returned early with no tokens. Yield the user-facing block message.
+                try:
+                    rail_result = _guardrail_task.result() if _guardrail_task.done() else None
+                except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)